src/List/Extra.elm

```diff
@@ -1863,25 +1863,32 @@ skipped and not appear in any groups.
 
 -}
 groupsOfWithStep : Int -> Int -> List a -> List (List a)
-groupsOfWithStep size step xs =
-    let
-        thisGroup =
-            List.take size xs
+groupsOfWithStep size step list =
+    if size <= 0 || step <= 0 then
+        []
 
-        xs_ =
-            List.drop step xs
+    else
+        let
+            go : List a -> List (List a) -> List (List a)
+            go xs acc =
+                if List.isEmpty xs then
+                    List.reverse acc
 
-        okayArgs =
-            size > 0 && step > 0
+                else
+                    let
+                        thisGroup =
+                            List.take size xs
 
-        okayLength =
-            size == List.length thisGroup
-    in
-    if okayArgs && okayLength then
-        thisGroup :: groupsOfWithStep size step xs_
+                        rest =
+                            List.drop step xs
```

Comment:
You could move this definition to inside the if (then) condition. That way, we won't compute the rest if `size /= List.length thisGroup`.

Reply:
Done in fb47d4a, thanks!